# Cálculo - Tarefa 02

1- Calcule de forma numérica a integral da função $x^3$ avaliada entre os pontos $1$ e $2$. Identifique o número de retângulos que você usou.  
2- Calcule essa área de forma analítica.

In [11]:
import pandas as pd
import numpy as np

In [13]:
f = lambda x: x**3

def calcular_area(a:int, b:int, func:callable, n_ret:int) -> float:
    delta = (b - a) / n_ret
    x_cols = np.linspace(a, b - delta, n_ret) + delta / 2
    y_cols = func(x_cols)
    
    print(f'Número de retângulos = {n_ret}')
    print(f'Delta = {delta}')
    
    return sum(delta * y_cols)

In [15]:
calcula_area(1, 2, f, 50)

3.749850000000001

$$\int_{a}^{b} f(x) dx = F(b) - F(a)$$

$$ f(x) = x^3 \rightarrow F(x) = \frac{1}{4} * x^4 + C,$$

$$\int_{1}^{2} x^3 dx = F(2) - F(1) = \left(\frac{1}{4} * 2^4\right) - \left(\frac{1}{4} * 1^4\right) = \frac{16}{4} - \frac{1}{4} = \frac{15}{4} = \textbf{3.75}$$